circadian/readers.py

```diff
@@ -1,49 +1,297 @@
+        if inplace:
+            for key, value in metadata.items():
+                self._obj.attrs[key] = value
+        else:
+            obj = self._obj.copy()
+            for key, value in metadata.items():
+                obj.attrs[key] = value
+            return obj
+
+    def plot(self, 
+             name: str, # name of the wereable data to plot (one of steps, heartrate, wake, light_estimate, or activity)
+             ax=None, # matplotlib axes
+             *args, # arguments to pass to matplotlib.pyplot.plot
+             **kwargs # keyword arguments to pass to matplotlib.pyplot.plot
+             ):
+        if name not in VALID_WEREABLE_STREAMS:
+            raise AttributeError(f"Name must be one of: {VALID_WEREABLE_STREAMS}.")
+        if ax is None:
+            ax = self._obj.plot(x='datetime', y=name, *args, **kwargs)
+        else:
+            ax = self._obj.plot(x='datetime', y=name, ax=ax, *args, **kwargs)
+        return ax
+
+# %% ../nbs/api/05_readers.ipynb 11
+# TODO: currently not in use because `jsonschema.validate` is very slow
+JSON_SCHEMA = {
+    type: "object",
+    "properties": {
+            "steps": {
+                "type": "array",
+                "items": { "type": "object", "properties": { 
+                    "start": { "type": "number" },
+                    "end": { "type": "number" },
+                    "steps": { "type": "number" } } },
+                "minItems": 1
+            },
+            "wake": {
+                "type": "array",
+                "items": { "type": "object", "properties": { 
+                    "start": { "type": "number" },
+                    "end": { "type": "number" },
+                    "wake": { "type": "number" } } }
+            },
+            "heartrate": {
+                "type": "array",
+                "items": { "type": "object", "properties": { 
+                    "timestamp": { "type": "number" },
+                    "heartrate": { "type": "number" },
+            } } },
+    },
+    "required": ["steps", "wake", "heartrate"]
+}
 
+# %% ../nbs/api/05_readers.ipynb 12
+def load_json(filepath: str, # path to file
+              metadata: Dict[str, str] = None, # metadata containing data_id, subject_id, or other_info
+              ) -> Dict[str, pd.DataFrame]: # dictionary of wereable dataframes, one key:value pair per wereable data stream
+    "Create a dataframe from a json containing a single or multiple streams of wereable data"
+    # validate inputs
+    if not isinstance(filepath, str):
+        raise AttributeError("Filepath must be a string.")
+    if metadata is not None:
+        WereableAccessor._validate_metadata(metadata)
+    # load json
+    jdict = json.load(open(filepath, 'r'))
+    # jsonschema.validate(jdict, schema=JSON_SCHEMA) # TODO: check for a better option, this is VERY slow
+    # check that keys are valid
+    for key in jdict.keys():
+        if key not in VALID_WEREABLE_STREAMS:
```

Comment:
Notes on common wearable data types: 

1) "Apple-type data"
- Not fixed interval 
Steps:
- Missing data is hard to distinguish, in the case of steps, from inactivity
- Apple steps: Start, end, step count. Duration is not constant, nor are the starts (by default) evenly space
- E.g. Start: 139489025, End: 139489100, Steps: 4
Heart rate:
- All heart rate measurements are instantaneous
- Not evenly spaced
- No signal = Person was not wearing the watch
- E.g. Start: 124981935, Value: 68 beats per minute

2) All others
- Evenly spaced data with a fixed "chunk" size
- Fitbit, Actiwatch, you name it: Most of these return values for activity like Between 12:00 and 12:01, you had 30 "counts"
- Or Between 12:00 and 12:01, your average heart rate was 68 bpm


**An important thing to keep in mind when dealing with Apple type data is handling missingness; e.g. you don't want to interpolate over a big gap**